pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -358,12 +358,26 @@ func (ds *dockerService) getIP(podSandboxID string, sandbox *dockertypes.Contain
 	return ""
 }
 
+// Returns the inspect container response, the sandbox metadata, and network namespace mode
+func (ds *dockerService) getPodSandboxDetails(podSandboxID string) (*dockertypes.ContainerJSON, *runtimeapi.PodSandboxMetadata, runtimeapi.NamespaceMode, error) {
+	resp, err := ds.client.InspectContainer(podSandboxID)
+	if err != nil {
+		return nil, nil, runtimeapi.NamespaceMode_NODE, err
+	}
+
+	metadata, err := parseSandboxName(resp.Name)
+	if err != nil {
+		return nil, nil, runtimeapi.NamespaceMode_NODE, err
+	}
+
+	return resp, metadata, networkNamespaceMode(resp), nil
```

Comment:
given that you have to do additional work on the namespaceMode later to get the answer you want, maybe you should just remove that return argument, and then the caller can just do

    hostNetwork = (networkNamespaceMode(resp) == runtimeapi.NamespaceMode_NODE)

?

Reply:
@danwinship done, PTAL thanks!